src/credentials/CHIPCertFromX509.cpp

```diff
@@ -160,12 +156,26 @@ static CHIP_ERROR ConvertDistinguishedName(ASN1Reader & reader, TLVWriter & writ
                             attrOID == chip::ASN1::kOID_AttributeType_ChipICAId ||
                             attrOID == chip::ASN1::kOID_AttributeType_ChipRootId)
                         {
+                            if (attrOID == chip::ASN1::kOID_AttributeType_ChipNodeId)
+                            {
+                                VerifyOrReturnError(IsOperationalNodeId(chipAttr), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Suggest replacing *ALL* `CHIP_ERROR_INVALID_ARGUMENT` related to subjects to `CHIP_ERROR_WRONG_CERT_SUBJECT`

Reply:
Replaced `CHIP_ERROR_WRONG_CERT_SUBJECT` with `CHIP_ERROR_WRONG_CERT_DN` and using it where appropriate.